chain/client/src/config_updater.rs

```diff
@@ -0,0 +1,53 @@
+use near_chain_configs::UpdateableClientConfig;
+use near_dyn_configs::{UpdateableConfigLoaderError, UpdateableConfigs};
+use std::sync::Arc;
+use tokio::sync::broadcast::Receiver;
+
+#[derive(Debug)]
+pub enum ClientConfigUpdateError {}
+
+/// Manages updating the config encapsulating.
+pub struct ConfigUpdater {
+    /// Receives config updates while the node is running.
+    rx_config_update: Receiver<Result<UpdateableConfigs, Arc<UpdateableConfigLoaderError>>>,
+
+    /// Represents the latest Error of reading the dynamically reloadable configs.
+    updateable_configs_error: Option<Arc<UpdateableConfigLoaderError>>,
+}
+
+impl ConfigUpdater {
+    pub fn new(
+        rx_config_update: Receiver<Result<UpdateableConfigs, Arc<UpdateableConfigLoaderError>>>,
+    ) -> Self {
+        Self { rx_config_update, updateable_configs_error: None }
+    }
+
+    pub fn update_if_needed(&mut self, update_client_config_fn: &dyn Fn(UpdateableClientConfig)) {
```

Comment:
nit: maybe rename this to 'try_update' and move the comments to outside of the function as the documentation for this function